Sources/Document/CRDT/CRDTArray.swift

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2022 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/**
+ * `CRDTArray` represents Array data type containing logical clocks.
+ *
+ */
+class CRDTArray: CRDTContainer {
+    private var elements: RGATreeList
+
+    init(createdAt: TimeTicket, elements: RGATreeList = RGATreeList()) {
+        self.elements = elements
+        super.init(createdAt: createdAt)
+    }
+
+    /**
+     * `subPath` returns subPath of JSONPath of the given `createdAt` element.
+     */
+    override func subPath(createdAt: TimeTicket) throws -> String {
+        return try self.elements.subPath(createdAt: createdAt)
+    }
+
+    /**
+     * `purge` physically purge child element.
+     */
+    override func purge(element: CRDTElement) throws {
+        try self.elements.purge(element)
+    }
+
+    /**
+     * `insert` inserts the given element after the given previous element.
+     */
+    func insert(value: CRDTElement, afterCreatedAt: TimeTicket) throws {
+        try self.elements.insert(value, afterCreatedAt: afterCreatedAt)
+    }
+
+    /**
+     * `move` moves the given `createdAt` element after the `prevCreatedAt`.
+     */
+    func move(createdAt: TimeTicket, afterCreatedAt: TimeTicket, executedAt: TimeTicket) throws {
+        try self.elements.move(createdAt: createdAt, afterCreatedAt: afterCreatedAt, executedAt: executedAt)
+    }
+
+    /**
+     * `get` returns the element of the given createAt.
+     */
+    func get(createdAt: TimeTicket) throws -> CRDTElement {
+        guard let node = try? self.elements.get(createdAt: createdAt), node.isRemoved() == false else {
+            let log = "can't find the given node: \(createdAt)"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        return node
+    }
+
+    /**
+     * `get` returns the element of the given index.
+     */
+    func get(index: Int) throws -> CRDTElement {
```

Comment:
get(createdAt:) returns non-removed CRDTElements but this returns all CRDTElements.
We plan to fix it in the future.